modules/nf-core/sourmash/compare/main.nf

```diff
@@ -0,0 +1,39 @@
+process SOURMASH_COMPARE {
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    path signatures
```

Comment:
Consider having a meta and prefix anyway (e.g. different batches/'databases' of samples?)

Reply:
I think that's a great idea -- I think the most common use cases for having different sourmash compare outputs would be:
1. same set of samples, but comparing and different k-mer sizes. By default, `sourmash sketch` emits sketches with ksizes of 21, 31, and 51. Comparing at higher kmer sizes usually reduces similarity between samples.
2. same set of samples, but comparing with or without abundance tracking. When abundances are used, jaccard is used for the compare matrix. When abundances are not used, cosine is used.

I don't know how to integrate meta and prefix for this type of information...I thought meta and prefix were associated with sample files prefixes? e.g. everything before `.fastq.gz` in a file.

finally, a less common use case would be different sets of samples -- e.g. 9 samples from a gut microbiome get compared to each other, while 9 samples from skin get compared to each other. I think this might be less common because you could compare all 18 samples at one time and later subset to the bits you care about.

Reply:
@jfy133 would you be willing to weigh in on how I might approach this?

Reply:
Sorry @taylorreiter I wrote a whole review this morning (see below) and GitHub diddn't bother to submit it (so it was stuck in pending) :facepalm: